src/vue/forms/AssetUpdateForm.vue

```diff
@@ -0,0 +1,365 @@
+
+import { Sdk } from '@/sdk'
+import { base, ASSET_POLICIES } from '@tokend/js-sdk'
+
+import moment from 'moment'
+
+import { mapGetters } from 'vuex'
+import { vuexTypes } from '@/vuex'
+
+import { required, maxLength } from '@validators'
+
+const STEPS = {
+  information: {
+    number: 1,
+    titleId: 'asset-form.information-step',
+  },
+  advanced: {
+    number: 2,
+    titleId: 'asset-form.advanced-step',
+  },
+}
+const ASSET_CREATION_REQUEST_ID = '0'
+const EVENTS = {
+  update: 'update',
+}
+const EMPTY_DOCUMENT = {
+  mime_type: '',
+  name: '',
+  key: '',
+}
+
+const NAME_MAX_LENGTH = 255
+
+export default {
+  name: 'asset-update-form',
+  components: {
+    FormStepper,
+    Loader,
+  },
+  mixins: [FormMixin],
+  props: {
+    assetForUpdate: { type: String, required: true },
+  },
+
+  data: _ => ({
+    form: {
+      information: {
+        name: '',
+        logo: null,
+        policies: [],
+      },
+      advanced: {
+        terms: null,
+      },
+    },
+    request: {},
+    isLoaded: false,
+    isLoadingFailed: false,
+    currentStep: 1,
+    STEPS,
+    DOCUMENT_TYPES,
+    ASSET_POLICIES,
+    NAME_MAX_LENGTH,
+  }),
+
+  validations () {
+    return {
+      form: {
+        information: {
+          name: {
+            required,
+            maxLength: maxLength(NAME_MAX_LENGTH),
+          },
+        },
+      },
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      account: vuexTypes.account,
+    }),
+
+    assetRequestOpts () {
+      const requestId = this.request.id || ASSET_CREATION_REQUEST_ID
+      const logo = this.form.information.logo
+      const terms = this.form.advanced.terms
+
+      return {
+        requestID: requestId,
+        code: this.assetForUpdate,
+        policies: this.form.information.policies.reduce((a, b) => (a | b), 0),
+        details: {
+          name: this.form.information.name,
+          logo: logo ? logo.getDetailsForSave() : EMPTY_DOCUMENT,
+          terms: terms ? terms.getDetailsForSave() : EMPTY_DOCUMENT,
+        },
+      }
+    },
+  },
+
+  async created () {
+    try {
+      this.request = await this.getAssetRequestForUpdate(this.assetForUpdate)
+      this.populateForm()
+      this.isLoaded = true
+    } catch (e) {
+      this.isLoadingFailed = true
+      ErrorHandler.processWithoutFeedback(e)
+    }
+  },
+
+  methods: {
+    populateForm () {
+      this.form = {
+        information: {
+          name: this.request.assetName,
+          logo: this.request.logo.key
+            ? new DocumentContainer(this.request.logo)
+            : null,
+          policies: this.request.policies,
+        },
+        advanced: {
+          terms: this.request.terms.key
+            ? new DocumentContainer(this.request.terms)
+            : null,
+        },
+      }
+    },
+
+    async getAssetRequestForUpdate (assetCode) {
+      try {
+        const { data } = await Sdk.horizon.request.getAllForAssets({
```

Comment:
you have the bug here:

* the `/request/assets` endpoint is pageable, and by default is ordered by ID in `asc` direction
* ID's is assigned by just incrementing the previous one
* the default limit for records on 1 page is `10`

so if you have `> 10` requests, you'll suck, because the latest request will be on some of the next pages and won't come to you

Reply:
Because of the fact that you have both `assetCreate` and `assetUpdate` requests, the only way to completely fix the issue you need to fetch all pages.

Instead of doing this, you can just increase the page limit and ask for descending ordering for the records. Also you can ask separately for `pending` and `approved` requests if you need both. It won't completely fix the issue, but will reduce the probability of that happen

Reply:
Also by doing that you can remove a lot of code below, because you can assume that `requests` array is already sorted and it's `0` element is the latest request that was created

Reply:
Everything is actual for `AssetCreateForm` too

Reply:
Updated. Using `request` as a prop for `AssetCreateForm`, increased limit for `pending` and `rejected` requests for `AssetUpdateForm`.

Restored and updated `fromAssetRecord` method for asset request records.